Stop accessing func._schema in _python_dispatch.correct_storage_aliasing

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #161455
* #161432
* #161329
* #161328
* #161317
* #161315
* #161308
* #161304
* __->__ #161292
* #161301

func._schema is a pybind, accessing the arguments/returns is expensive, we have no reason to do it anyway, and even though #161301 makes accessing the arguments/returns less expensive, this still seems to improve performance.

swolchok committed:
Stop accessing func._schema in _python_dispatch.correct_storage_aliasing

func._schema is a pybind, accessing the arguments/returns is expensive, and we have no reason to do it anyway.

[ghstack-poisoned]

swolchok committed:
Update on "Stop accessing func._schema in _python_dispatch.correct_storage_aliasing"


func._schema is a pybind, accessing the arguments/returns is expensive, and we have no reason to do it anyway.

[ghstack-poisoned]

swolchok committed:
Update on "Stop accessing func._schema in _python_dispatch.correct_storage_aliasing"


func._schema is a pybind, accessing the arguments/returns is expensive, we have no reason to do it anyway, and even though #161301 makes accessing the arguments/returns less expensive, this still seems to improve performance.

[ghstack-poisoned]

swolchok committed:
Update on "Stop accessing func._schema in _python_dispatch.correct_storage_aliasing"


func._schema is a pybind, accessing the arguments/returns is expensive, we have no reason to do it anyway, and even though #161301 makes accessing the arguments/returns less expensive, this still seems to improve performance.

[ghstack-poisoned]

swolchok committed:
Update on "Stop accessing func._schema in _python_dispatch.correct_storage_aliasing"


func._schema is a pybind, accessing the arguments/returns is expensive, we have no reason to do it anyway, and even though #161301 makes accessing the arguments/returns less expensive, this still seems to improve performance.

[ghstack-poisoned]

malfet reviewed (approved):
I think PR title is a bit misleading, `[BE] Use enumerate in alias_non_inplace_storage` sounds better to me

swolchok committed:
rebase on "Stop accessing func._schema in _python_dispatch.correct_storage_aliasing"


func._schema is a pybind, accessing the arguments/returns is expensive, we have no reason to do it anyway, and even though #161301 makes accessing the arguments/returns less expensive, this still seems to improve performance.

[ghstack-poisoned]

swolchok committed:
Update on "Stop accessing func._schema in _python_dispatch.correct_storage_aliasing"


func._schema is a pybind, accessing the arguments/returns is expensive, we have no reason to do it anyway, and even though #161301 makes accessing the arguments/returns less expensive, this still seems to improve performance.

[ghstack-poisoned]
